packages/docsearch-react/src/utils/removeHighlightTags.ts

```diff
@@ -6,17 +6,17 @@ const regexHasHighlightTags = RegExp(regexHighlightTags.source);
 export function removeHighlightTags(
   hit: DocSearchHit | InternalDocSearchHit
 ): string {
-  if (
-    !(hit as InternalDocSearchHit).__docsearch_parent &&
-    !hit._highlightResult
-  ) {
+  const internalDocSearchHit = hit as InternalDocSearchHit;
+
+  if (!internalDocSearchHit.__docsearch_parent && !hit._highlightResult) {
     return hit.hierarchy.lvl0;
   }
 
-  const { value } = hit._highlightResult
-    ? hit._highlightResult.hierarchy.lvl0
-    : (hit as InternalDocSearchHit).__docsearch_parent!._highlightResult
-        .hierarchy.lvl0;
+  const { value } =
+    (internalDocSearchHit.__docsearch_parent
+      ? internalDocSearchHit.__docsearch_parent?._highlightResult?.hierarchy
+          ?.lvl0
+      : hit._highlightResult?.hierarchy?.lvl0) || '';
```

Comment:
```suggestion
      : hit._highlightResult?.hierarchy?.lvl0) || {};
```

you're trying to restructure from '' otherwise right?

Reply:
are there any tests we can add this case to?

Reply:
> you're trying to destructure from '' otherwise right?

Both returns `undefined` but yours is more correct indeed, I wonder if we should default it to something like `Documentation` to avoid empty `lvl0` like this:

![Screenshot 2021-08-19 at 17 11 46](https://user-images.githubusercontent.com/20689156/130094079-d1631f25-f329-4164-9d35-7e328d14a545.png)


> are there any tests we can add this case to?

There's no tests other than cypress on this repo yet. Since we're planning to re-write the component after the Reshape API has been introduced, I'm not sure if I should add some now

Reply:
I'm not sure adding a static item when lvl0 is missing makes sense, the UI would have a lot of useless info

Reply:
It's not a big deal anyway and I've seen this happen only twice in the past year